Sources/LiveKit/Core/SignalClient.swift

```diff
@@ -84,20 +88,30 @@ internal class SignalClient: MulticastDelegate<SignalClientDelegate> {
         }
 
         latestJoinResponse = nil
+        queue.removeAll()
     }
 }
 
 // MARK: - Private
 
 private extension SignalClient {
 
-    func sendRequest(_ request: Livekit_SignalRequest) -> Promise<Void> {
+    // send request or enqueue while reconnecting
+    func sendRequest(_ request: Livekit_SignalRequest, enqueueIfReconnecting: Bool = true) -> Promise<Void> {
+
+        guard !(connectionState.isReconnecting && request.canQueue() && enqueueIfReconnecting) else {
```

Comment:
when reconnection is finished, I think we'd still have to enqueue requests *as long as* the queue is not empty.

this is to ensure ordering is kept with the signal messages. i.e. replacing connectionState.isReconnecting => connectionState.IsReconnecting || !queue.isNotEmpty

Reply:
After reconnection is finished, and sending out the queued requests, clearing the queue wouldn't work ?

Reply:
yeah it should.. I'm thinking about this case:

1. one thread is flushing the queue after reconnecting, while that's happening
2. user initiates an action that causes a signal event to be sent

would 2 be queued until after the flush is complete? or would it be sent as it comes in?

Reply:
Good point about mutex, i'll make some changes to have peace of mind.

Reply:
I have fixed this so `requestQueue` cannot be modified (promise will wait) while a thread is flushing the queue.